lib/ok_then/result.ex

```diff
@@ -493,6 +493,93 @@ defmodule OkThen.Result do
     end)
   end
 
+  @doc """
+  If `result` is tagged `:ok`, passes the wrapped value into the provided function. If
+  `check_function` returns `true` the `result` is returned unchanged. Otherwise, returns `:none`.
+
+  Equivalent to `tagged_ensure(result, :ok, check_function)`. See `tagged_ensure/3`.
+
+  ## Examples
+
+      iex> {:ok, "hello"} |> Result.ensure(&String.length(&1) == 5)
+      {:ok, "hello"}
+
+      iex> {:ok, "hello"} |> Result.ensure(&String.length(&1) == 0)
+      :none
+
+      iex> :some |> Result.ensure(&String.length(&1) == 0)
+      :some
+
+      iex> :error |> Result.ensure(&String.length(&1) == 0)
+      :error
+
+      iex> nil |> Result.ensure(&String.length(&1) == 0)
+      nil
+  """
+  @spec ensure(result_input(), (any() -> boolean())) :: result_input()
+  def ensure(result, check_function) when is_function(check_function, 1),
+    do: tagged_ensure(result, :ok, check_function)
+
+  @doc """
+  If `result` is tagged `:error`, passes the wrapped value into the provided function. If
+  `check_function` returns `true` the `result` is returned unchanged. Otherwise, returns `:none`.
+
+  Equivalent to `tagged_ensure(result, :error, check_function)`. See `tagged_ensure/3`.
+
+  ## Examples
+
+      iex> {:error, "hello"} |> Result.error_ensure(&String.length(&1) == 5)
+      {:error, "hello"}
+
+      iex> {:error, "hello"} |> Result.error_ensure(&String.length(&1) == 0)
+      :none
+
+      iex> :some |> Result.error_ensure(&String.length(&1) == 0)
+      :some
+
+      iex> :ok |> Result.error_ensure(&String.length(&1) == 0)
+      :ok
+
+      iex> nil |> Result.error_ensure(&String.length(&1) == 0)
+      nil
+  """
+  @spec error_ensure(result_input(), (any() -> boolean())) :: result_input()
+  def error_ensure(result, check_function) when is_function(check_function, 1),
+    do: tagged_ensure(result, :error, check_function)
+
+  @doc """
+  If `result` is tagged with the specified `tag` atom, passes the wrapped value into the provided
+  function. If `check_function` returns `true` the `result` is returned unchanged. Otherwise,
```

Comment:
Technically it's just if it returns a truthy value, right?

Reply:
Yes, quite right :+1: